app/models/filter.rb

```diff
@@ -176,7 +176,7 @@ def build_taxonomy_search
       self.taxonomy_search = nil
     else
       taxonomies = [orgs, locs].reject { |t| t.blank? }
-      self.taxonomy_search = taxonomies.join(' and ')
+      self.taxonomy_search = taxonomies.join(' and ') unless taxonomies.empty?
```

Comment:
This appears to have been the root cause. I was able to create filters with an empty string as the value of `taxonomy_search` even though it should be set nil

Reply:
Oh right, because `enforce_inherited_taxonomies` goes after `nilify_empty_serches`, because it's `before_save`.
What about moving the `enforece_inherited_taxonomies` before `nilify_empty_searches`?
I'd be even ok with having strict validation for not being empty string :thinking: